pkg/mobile/metrics/fh-sync-server.go

```diff
@@ -0,0 +1,347 @@
+		if err == nil {
+			ssMetrics = append(ssMetrics, &metric{Type: "mongodb_operation_time_doReadDatasetClientWithRecordsUseCache_ms", XValue: now.Format("2006-01-02 15:04:05"), YValue: doReadDatasetClientWithRecordsUseCache})
+		}
+		doFindAndDeleteUpdate, err := stringToInt64(stats.Metrics.MongodbOperationTime.DoFindAndDeleteUpdate.Current)
+		if err == nil {
+			ssMetrics = append(ssMetrics, &metric{Type: "mongodb_operation_time_doFindAndDeleteUpdate_ms", XValue: now.Format("2006-01-02 15:04:05"), YValue: doFindAndDeleteUpdate})
+		}
+
+		// sync worker stats
+		ssMetrics = append(ssMetrics, &metric{Type: "sync_worker_queue_count", XValue: now.Format("2006-01-02 15:04:05"), YValue: stats.Metrics.JobQueueSize.SyncWorker.Current})
+		syncProcessTime, err := stringToInt64(stats.Metrics.JobProcessTime.SyncWorker.Current)
+		if err == nil {
+			ssMetrics = append(ssMetrics, &metric{Type: "sync_worker_process_time_ms", XValue: now.Format("2006-01-02 15:04:05"), YValue: syncProcessTime})
+		}
+
+		// pending worker stats
+		ssMetrics = append(ssMetrics, &metric{Type: "pending_worker_queue_count", XValue: now.Format("2006-01-02 15:04:05"), YValue: stats.Metrics.JobQueueSize.PendingWorker.Current})
+		pendingProcessTime, err := stringToInt64(stats.Metrics.JobProcessTime.PendingWorker.Current)
+		if err == nil {
+			ssMetrics = append(ssMetrics, &metric{Type: "pending_worker_process_time_ms", XValue: now.Format("2006-01-02 15:04:05"), YValue: pendingProcessTime})
+		}
+
+		// ack worker stats
+		ssMetrics = append(ssMetrics, &metric{Type: "ack_worker_queue_count", XValue: now.Format("2006-01-02 15:04:05"), YValue: stats.Metrics.JobQueueSize.AckWorker.Current})
+		ackProcessTime, err := stringToInt64(stats.Metrics.JobProcessTime.AckWorker.Current)
+		if err == nil {
+			ssMetrics = append(ssMetrics, &metric{Type: "ack_worker_process_time_ms", XValue: now.Format("2006-01-02 15:04:05"), YValue: ackProcessTime})
+		}
+	}
+
+	return ssMetrics, nil
+}
+
+func (ss *FhSyncServer) getStats(host string) (*statsResponse, error) {
+	statsURL := fmt.Sprintf("%s/sys/info/stats", host)
+	requester := ss.requestBuilder.Insecure(true).Build()
+	res, err := requester.Get(statsURL)
+	if err != nil {
+		return &statsResponse{}, err
+	}
+	defer res.Body.Close()
+
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return &statsResponse{}, errors.Wrap(err, "error reading stats response body")
+	}
+	ss.logger.Debugf("raw stats response %v", string(data))
+
+	var stats statsResponse
+	decoder := json.NewDecoder(bytes.NewBuffer(data))
+	if err := decoder.Decode(&stats); err != nil {
+		return &statsResponse{}, errors.Wrap(err, "failed to decode stats response")
+	}
+	ss.logger.Debugf("decoded stats response %v", stats)
+
+	return &stats, nil
+}
+
+func stringToInt64(val string) (int64, error) {
+	// string expected in format "00.00ms"
```

Comment:
do we want to test the format with a regex or maybe test the length of the split string?

Reply:
Yeah, lemme put something together that's  more robust at parsing the number